spec/controllers/hq/sponsorships_controller_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+RSpec.describe Hq::SponsorshipsController, type: :controller do
+  let(:sponsorship) {create :sponsorship, active: true}
+
+  before :each do
+    sign_in instance_double(AdminUser)
+  end
+
+  specify "#inactivate" do
+    expect(Sponsorship).to receive(:find).and_return(sponsorship)
+    put :inactivate, id: sponsorship.id, sponsorship: {end_date: Date.current}
+
+    expect(sponsorship.active).to be false
```

Comment:
Sorry, I should have made myself clearer from the beginning: We should not be testing that the sponsorship is, in fact, inactivated, just as we don't test that the sponsorship actually gets destroyed below, i.e. L14 should not be here at all. Those are the responsibilities of the respective services that are exercised in their own specs. All we test here are the messages that the controller sends and its response to the replies that come back.

I am suggesting a spec along the lines of https://github.com/AgileVentures/osra/blob/develop/spec/controllers/admin/sponsorships_controller_spec.rb.